### PR TITLE
CI cache the datasets to avoid DL failures

### PR DESCRIPTION
**Pull request by zaccharieramzi**

### CODE REVIEW

**Review by tomMoral — APPROVED** [review]
LGTM!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -98,6 +98,19 @@ jobs:
...
+        env:
+          cache-name: cache-datasets
+        with:
+          path: |
+            data
+            ~/tensorflow_datasets
+          key: build-${{ env.cache-name }}-${{ github.ref }}
+          restore-keys: |
+            build-${{ env.cache-name }}-
+
```
Comment (tomMoral) [inline .github/workflows/main.yml:113]: You have an identation issue in the yaml file.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -98,6 +98,19 @@ jobs:
...
+        uses: actions/cache@v3
+        env:
+          cache-name: cache-datasets
+        with:
+          path: |
+            data
+            ~/tensorflow_datasets
+          key: build-${{ env.cache-name }}-${{ github.ref }}
+          restore-keys: |
+            build-${{ env.cache-name }}-
```
Comment (tomMoral) [inline .github/workflows/main.yml:112]:
```suggestion
     uses: actions/cache@v3
     env:
       cache-name: cache-datasets
     with:
       path: |
         data
         ~/tensorflow_datasets
       key: build-${{ env.cache-name }}-${{ github.ref }}
       restore-keys: |
         build-${{ env.cache-name }}-
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (zaccharieramzi) [reply .github/workflows/main.yml]: yup corrected

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -98,6 +98,19 @@ jobs:
...
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-datasets
+      with:
+        path: |
+          data
+          ~/tensorflow_datasets
+        key: build-${{ env.cache-name }}-${{ github.ref }}
+        restore-keys: |
+          build-${{ env.cache-name }}-
```
Comment (tomMoral) [inline .github/workflows/main.yml:112]: I think we can even simplify as it should not change between the PR, the datasets are pretty constants.
```suggestion
        key: build-${{ env.cache-name }}
        restore-keys: |
          build-${{ env.cache-name }}
```

**Comment by tomMoral** [comment]
Let's wait for the CI to fully run, so we can see if the cache works as expected when we commit the sugestion

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -98,6 +98,19 @@ jobs:
...
+      uses: actions/cache@v3
+      env:
+        cache-name: cache-datasets
+      with:
+        path: |
+          data
+          ~/tensorflow_datasets
+        key: build-${{ env.cache-name }}
+        restore-keys: |
+          build-${{ env.cache-name }}
```
Comment (tomMoral) [inline .github/workflows/main.yml:112]: We should add the same cache in `test-benchmark`

Comment (zaccharieramzi) [reply .github/workflows/main.yml:123]: yes, done

**Pull request merged**